superset/db_engine_specs/postgres.py

```diff
@@ -45,6 +59,17 @@ class PostgresBaseEngineSpec(BaseEngineSpec):
     engine = ""
     engine_name = "PostgreSQL"
 
+    column_type_mappings = (
+        (
+            re.compile(r"^double precision", re.IGNORECASE),
+            DOUBLE_PRECISION(),
+            GenericDataType.NUMERIC,
+        ),
+        (re.compile(r"^array.*", re.IGNORECASE), ARRAY(), utils.GenericDataType.STRING),
```

Comment:
It appears `ARRAY()` needs to be instantiated with a type: https://docs.sqlalchemy.org/en/14/dialects/postgresql.html#sqlalchemy.dialects.postgresql.ARRAY I think you'll need to use a callback to match the type of the array and then pass that as `item_type` (see `VARCHAR` in Presto spec).